components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWALocalAuthenticator.java

```diff
@@ -0,0 +1,145 @@
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authenticator.iwa.internal.IWAServiceDataHolder;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+/**
+ * IWALocalAuthenticator authenticates a user from a Kerberos Token (GSS Token) sent by a pre-registered KDC
+ * (ie. the active directory). Currently we only authenticate users from the Primary user store of the super tenant
+ * with this authenticator
+ */
+public class IWALocalAuthenticator extends AbstractIWAAuthenticator implements
+        LocalApplicationAuthenticator {
+
+    public static final String AUTHENTICATOR_NAME = "IWALocalAuthenticator";
+    public static final String AUTHENTICATOR_FRIENDLY_NAME = "iwa-local";
+
+    //the following param of the request will be set once the request is processed by the IWAServlet
+    private static final long serialVersionUID = -713445365110141399L;
+    private static Log log = LogFactory.getLog(IWALocalAuthenticator.class);
+
+
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+        super.processAuthenticationResponse(request, response, context);
+
+        HttpSession session = request.getSession(false);
+        final String gssToken = (String) session.getAttribute(IWAConstants.GSS_TOKEN);
+
+        // get the authenticated username by processing the GSS Token
+        String authenticatedUserName = null;
+        try {
+            authenticatedUserName = getAuthenticatedUserFromToken(Base64.decode(gssToken));
+        } catch (GSSException e) {
+            throw new AuthenticationFailedException("Error extracting username from the GSS Token.", e);
+        }
+
+        if (IdentityUtil.isBlank(authenticatedUserName)) {
+            throw new AuthenticationFailedException("Authenticated user not found in GSS Token : "
+                    + authenticatedUserName);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Authenticated user from GSS Token : " + authenticatedUserName);
+            }
+
+        }
+
+        // remove the AD domain from the username
+        int index = authenticatedUserName.lastIndexOf("@");
+        authenticatedUserName = authenticatedUserName.substring(0, index);
```

Comment:
True. MultiTenantUtils class provides similar util methods. But here we are trying to extract the windows domain from the fully qualified username. So using getTenantAwareUsername() method didn't seem right. So added a Util method [1] to be used within the IWA component.

[1] https://github.com/wso2-extensions/identity-local-auth-iwa/blob/master/components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAAuthenticationUtil.java#L294